editor/src/clj/editor/color_dropper.clj

```diff
@@ -0,0 +1,174 @@
+                      y (+ mouse-y (* y-step pixel-size) (- pixel-center))]]
+          (when (in-bounds? view-node (+ x delta-x) (+ y delta-y))
+            (->> (.getColor pixel-reader (+ adjusted-x x-step) (+ adjusted-y y-step))
+                 (paint-pixel! graphics-context x y pixel-size))))
+
+        (doto graphics-context
+          (.strokeOval (- mouse-x radius) (- mouse-y radius) diameter diameter)
+          (highlight-pixel! (- mouse-x pixel-center) (- mouse-y pixel-center) pixel-size))))))
+
+(defn- apply-and-deactivate!
+  [view-node ^Popup popup pick-fn]
+  (.hide popup)
+  (pick-fn (g/node-value view-node :color))
+  (ui/user-data! (ui/main-scene) ::ui/refresh-requested? true))
+
+(defn- get-bounds []
+  (let [bounds (reduce (fn [bounds ^Screen screen]
+                         (let [{:keys [^double min-x ^double min-y ^double max-x ^double max-y]} bounds
+                               screen-bounds (.getBounds screen)]
+                           (assoc bounds
+                                  :min-x (cond-> ^double (.getMinX screen-bounds) min-x (min min-x))
+                                  :min-y (cond-> ^double (.getMinY screen-bounds) min-y (min min-y))
+                                  :max-x (cond-> ^double (.getMaxX screen-bounds) max-x (max max-x))
+                                  :max-y (cond-> ^double (.getMaxY screen-bounds) max-y (max max-y)))))
+                       {}
+                       (Screen/getScreens))
+        {:keys [^double min-x ^double min-y ^double max-x ^double max-y]} bounds]
+    (assoc bounds
+           :width (- max-x min-x)
+           :height (- max-y min-y))))
+
+(defn make-color-dropper! [graph]
+  (g/make-node! graph ColorDropper))
+
+(defn- create-popup! ^Popup
+  [x y width height]
+   (doto (Popup.)
+     (.setX x)
+     (.setY y)
+     (.setWidth width)
+     (.setHeight height)))
+
+(defn- activate!
+  [view-node pick-fn]
+  (let [bounds (get-bounds)
+        {:keys [min-x min-y width height]} bounds
+        popup (create-popup! min-x min-y width height)
+        canvas (Canvas. width height)
+        pane (doto (Pane.)
+               (.setCursor Cursor/NONE)
+               (.setPrefSize 1 1)
+               (.setStyle "-fx-background-color: transparent;"))]
+    (g/set-property! view-node :bounds bounds)
+    
+    (.add (.getContent popup) pane)
+    (.addListener (.focusedProperty popup) (ui/change-listener _ _ v (when-not v (.hide popup))))
+    
+    (doto popup
+      (.addEventHandler KeyEvent/ANY (ui/event-handler event (.hide popup)))
+      (.addEventHandler MouseEvent/MOUSE_MOVED (ui/event-handler event (mouse-move-handler! view-node canvas event)))
+      (.addEventHandler MouseEvent/MOUSE_PRESSED (ui/event-handler event (apply-and-deactivate! view-node popup pick-fn)))
+      (.show (ui/main-stage))
+      (.requestFocus))
+    
+    (capture! view-node)
+    (doto pane
+      (.setPrefSize width height)
+      (ui/add-child! canvas))))
```

Comment:
The order of the events here is important to minimize flickering and make this work consistently on linux+wayland.